src/webgl/loading.js

```diff
@@ -150,18 +149,37 @@ p5.prototype.loadModel = function(path) {
       failureCallback
     );
   } else if (fileType.match(/\.obj$/i)) {
+    const mtlFiles=[];
+    var mtlPath='';
+    var parsedMaterials={};
     this.loadStrings(
       path,
       async lines => {
-        const mtlLine = lines.find(line => line.startsWith('mtllib '));
-
-        // console.log(mtlPath);
+        const mtlPromises = lines
+          .map(line => line.trim().split(/\b\s+/))
+          .filter(tokens => tokens[0] === 'mtllib')
+          .map(async tokens => {
+            const objFileDir = path.substring(0, path.lastIndexOf('/'));
+            //because path.dirname does not exist error to get path of mtlfiles
+
+            // Replace the file name in the path with the MTL file name
+            //find a better way to resolve path
+            if(objFileDir){
+              mtlPath = objFileDir + '/' + tokens[1];
+            }
+            else{
+              mtlPath=tokens[1];
+            }
+            // Check if the MTL file has already been loaded
+            const existingMtl = mtlFiles.find(mtl => mtl.path === mtlPath);
```

Comment:
I think this is a bit of a race condition, since we only push to the array after `await`ing an asynchronous function. Maybe you can synchronously push an object that just includes the path, and then after `await`ing, add the materials to it? something like:
```js
const file = { path: mtlPath };
mtlFiles.push(file);
const parsedMaterials = await parseMtl(self, mtlPath)
file.materials = parsedMaterials;
```